templates/base_index.html

```diff
@@ -119,6 +119,17 @@
         } else {
           document.querySelector(successElement).classList.add("u-hide");
         }
+
+        let params = new URLSearchParams(window.location.search);
+        const utm_names = ["campaign", "source", "medium", "content", "term"];
+        for (let i = 0; i < utm_names.length; i++) {
+          var utm_fields = document.getElementsByName("utm_" + utm_names[i]);
+          for (let j = 0; j < utm_fields.length; j++) {
+            if (utm_fields[j]) {
+              utm_fields[j].value = params.get("utm_" + utm_names[i]);
+            }
+          }
+        }
```

Comment:
I don't see why this needs to be nested for loops, seems to me that the `utm_fields` variable will always contain one field.

I think the code should look something like this:

```
for (let i = 0; i < utm_names.length; i++) {
  let utm_field = document.getElementsByName("utm_" + utm_names[i])[0];

  if (utm_field) {
    utm_field.value = params.get("utm_" + utm_names[i]);
  }
}
```

Unless there's a posibility we'll have duplicate utm fields on a page. I'll leave that up to you to decide.

Reply:
Yeah I think it does make sense what you say. I believe this was in place to fix in all forms that had duplicated fields, but then that fix would mean one value gets overriden by the other. So you're right, maybe we should remove the second loop.